src/components/Comments/Comment.js

```diff
@@ -111,6 +129,61 @@ const Comment = ({
     return COLORS.balance_gray;
   };
 
+  const username = comment?.data?.anon_user_info_color_name
+    ? `Anonymous ${comment.data?.anon_user_info_emoji_name}`
+    : user?.data?.username;
+
+  const onPressDM = async () => {
+    try {
+      setLoading({...loading, loadingDm: true});
+      if (!comment?.data?.anon_user_info_color_name) {
+        const channelName = [username, profile?.myProfile?.username].join(',');
+        const selectedUser = {
+          user: {
+            name: channelName,
+            image: comment?.user?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+          }
+        };
+        const members = [comment?.user?.id, profile.myProfile.user_id];
+        await createSignChat(members, selectedUser);
+      } else {
+        await sendMessageDM(comment?.id, 'comment', 'SIGNED');
+      }
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDm: false});
+    }
+  };
+
+  const onPressDMAnon = async () => {
+    try {
+      setLoading({...loading, loadingDmAnon: true});
+      await sendMessageDM(comment?.id, 'comment', 'ANONYMOUS');
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDmAnon: false});
+    }
+  };
+
+  const dataSheet = [
+    {
+      id: 1,
+      name: loading.loadingDm ? 'Loading...' : `Message ${username}`,
+      icon: <SendDMBlack />,
+      onPress: onPressDM
+    },
+    {
+      id: 2,
+      name: loading.loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
+      icon: <SendDMAnonBlock />,
+      onPress: onPressDMAnon
+    }
+  ];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This section introduces two new functions `onPressDM` and `onPressDMAnon` for sending direct messages. The error handling here is minimal with just a console warning. Consider providing more robust error handling and user feedback. Also, the dataSheet array is created every time the component renders. If the dataSheet doesn't change, consider using useMemo to avoid unnecessary computations.
```diff
+ const username = comment?.data?.anon_user_info_color_name
+   ? `Anonymous ${comment.data?.anon_user_info_emoji_name}`
+   : user?.data?.username;
+ 
+ const onPressDM = async () => {
+   try {
+     setLoading({...loading, loadingDm: true});
+     if (!comment?.data?.anon_user_info_color_name) {
+       const channelName = [username, profile?.myProfile?.username].join(',');
+       const selectedUser = {
+         user: {
+           name: channelName,
+           image: comment?.user?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+         }
+       };
+       const members = [comment?.user?.id, profile.myProfile.user_id];
+       await createSignChat(members, selectedUser);
+     } else {
+       await sendMessageDM(comment?.id, 'comment', 'SIGNED');
+     }
+   } catch (e) {
+     console.warn(e);
+   } finally {
+     refSheet.current.close();
+     setLoading({...loading, loadingDm: false});
+   }
+ };
+ 
+ const onPressDMAnon = async () => {
+   try {
+     setLoading({...loading, loadingDmAnon: true});
+     await sendMessageDM(comment?.id, 'comment', 'ANONYMOUS');
+   } catch (e) {
+     console.warn(e);
+   } finally {
+     refSheet.current.close();
+     setLoading({...loading, loadingDmAnon: false});
+   }
+ };
+ 
+ const dataSheet = [
+   {
+     id: 1,
+     name: loading.loadingDm ? 'Loading...' : `Message ${username}`,
+     icon: <SendDMBlack />,
+     onPress: onPressDM
+   },
+   {
+     id: 2,
+     name: loading.loadingDmAnon ? 'Loading...' : `Message ${username} anonymously`,
+     icon: <SendDMAnonBlock />,
+     onPress: onPressDMAnon
+   }
+ ];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->